cfme/common/provider.py

```diff
@@ -187,27 +187,35 @@ def create(self, cancel=False, validate_credentials=True, check_existing=False,
             if cancel:
                 created = False
                 add_view.cancel.click()
-                cancel_text = ('Add of {} Provider was '
-                               'cancelled by the user'.format(self.string_name))
-
-                main_view.flash.assert_message(cancel_text)
-                main_view.flash.assert_no_error()
+                self._post_create_checks(main_view, cancel=True)
             else:
                 add_view.add.click()
-                if main_view.is_displayed:
-                    success_text = '{} Providers "{}" was saved'.format(self.string_name,
-                                                                        self.name)
-                    main_view.flash.assert_message(success_text)
-                else:
-                    add_view.flash.assert_no_error()
-                    raise AssertionError("Provider wasn't added. It seems form isn't accurately"
-                                         " filled")
+                self._post_create_checks(main_view, add_view)
 
         if validate_inventory:
             self.validate()
 
         return created
 
+    def _post_create_checks(self, main_view, add_view=None, cancel=False):
```

Comment:
**REQUIRED**
It's not needed. We plan to assert flash messages in the tests. Only `flash.assert_no_error()` will be kept.